tableauserverclient/server/endpoint/jobs_endpoint.py

```diff
@@ -32,21 +38,20 @@ def get(self, job_id=None, req_options=None):
         return jobs, pagination_item
 
     @api(version="3.1")
-    def cancel(self, job_id):
+    def cancel(self, job_id: str):
         id_ = getattr(job_id, "id", job_id)
         url = "{0}/{1}".format(self.baseurl, id_)
         return self.put_request(url)
 
     @api(version="2.6")
-    def get_by_id(self, job_id):
+    def get_by_id(self, job_id: str) -> JobItem:
         logger.info("Query for information about job " + job_id)
         url = "{0}/{1}".format(self.baseurl, job_id)
         server_response = self.get_request(url)
         new_job = JobItem.from_response(server_response.content, self.parent_srv.namespace)[0]
         return new_job
 
-    @api(version="2.6")
-    def wait_for_job(self, job_id, *, timeout=None):
+    def wait_for_job(self, job_id: str, *, timeout: Optional[T] = None) -> JobItem:
```

Comment:
* `job_id: str` annotation is incorrect. The code below explicitly also allows `JobItem` instances
* `timeout: Optional[T]`: why `T`?  Can't this just be `Optional[float]`